core/templates/pages/exploration-editor-page/modal-templates/exploration-modify-translations-modal.component.spec.ts

```diff
@@ -0,0 +1,173 @@
+        translations: {
+          content1: {
+            content_value: 'This is text one.',
+            content_format: 'html',
+            needs_update: false,
+          },
+          content2: {
+            content_value: 'This is text two.',
+            content_format: 'html',
+            needs_update: false,
+          },
+          content3: {
+            content_value: 'This is text three.',
+            content_format: 'html',
+            needs_update: false,
+          },
+        },
+      }),
+    };
+    spyOn(
+      entityBulkTranslationsBackendApiService,
+      'fetchEntityBulkTranslationsAsync'
+    ).and.returnValue(
+      Promise.resolve({
+        hi: {
+          entityId: 'entity1',
+          entityType: 'exploration',
+          entityVersion: 5,
+          languageCode: 'hi',
+          translationMapping: {
+            content1: {
+              translation: '<p>This is content 1.</p>',
+              dataFormat: 'html',
+              needsUpdate: true,
+            },
+            content4: {
+              translation: '<p>This is content 4.</p>',
+              dataFormat: 'html',
+              needsUpdate: false,
+            },
+          },
+        },
+      })
+    );
+  });
+
+  it('should check component is initialized', () => {
+    expect(component).toBeDefined();
+  });
+
+  describe('when initializing content translations', () => {
+    it('should use latest translations from entity translations service', fakeAsync(() => {
+      spyOn(contextService, 'getExplorationId').and.returnValue('expId');
+      let expectedTranslation = TranslatedContent.createFromBackendDict({
+        content_value: 'This is text one.',
+        content_format: 'html',
+        needs_update: false,
+      });
+
+      component.contentId = 'content1';
+      component.ngOnInit();
+      tick();
+
+      expect(component.contentTranslations).toEqual({
+        hi: expectedTranslation,
+      });
+    }));
+
+    it('should use published translations when changes do not exist', fakeAsync(() => {
+      spyOn(contextService, 'getExplorationId').and.returnValue('expId');
+      let expectedTranslation = TranslatedContent.createFromBackendDict({
+        content_value: '<p>This is content 4.</p>',
+        content_format: 'html',
+        needs_update: false,
+      });
+
+      component.contentId = 'content4';
+      component.ngOnInit();
+      tick();
+
+      expect(component.contentTranslations).toEqual({
+        hi: expectedTranslation,
+      });
+    }));
+  });
+
+  it('should handle translations being removed', () => {
+    spyOn(contextService, 'getExplorationId').and.returnValue('expId');
+    spyOn(changeListService, 'getTranslationChangeList').and.returnValue([
+      {
+        cmd: 'remove_translations',
+        content_id: 'content4',
+      },
+    ]);
+    component.contentId = 'content4';
+    component.ngOnInit();
+    expect(component.contentTranslations).toEqual({});
```

Comment:
We cannot here, since the translations are both added and removed during the ngOnInit call. Unless you mean checking that they are empty at the start already?

Reply:
Yes, if the expected behaviour is that the variable is empty before the ngOnInit call then assert the same.

Reply:
Done!